### PR TITLE
Feat: custom sort

### LINKED ISSUE #4800: Allow custom sort options for conversations in the API
Support the following sort options in conversation API and filter APIs.

- **created_at**: Sort conversations based on created time. Allows agents to prioritize long-running conversations.
- **last_user_message_at**: Sort based on the last incoming message. Along with the unattended filter, this sort option allows agents to look at conversations where users are waiting for a long time.
- **last_activity_at**: Sort based on the last activity. This is the default sort right now.

### PR DESCRIPTION
**Pull request by tejaswinichile**
# Pull Request Template

## Description

Allow custom sort option for conversation and conversation-mentions
Send param[:sort_by] = 'lates'/'sort_on_created_at'/''last_user_message_at

**latest**: will give sort on last_activity_at
**created_at**: sort on conversation created at - oldest first
**last_user_message_at**: will sort on, conversation with the last message created at date. This sort is for all those conversations having a last message with no_reply. Conversations which has the last outgoing message, those will not be included in this sort.

Fixes #4800 

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Sort by latest
Sort by created-at
Sort by last user messaged at, for incoming message 

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by tejaswinichile** [comment]
https://www.loom.com/share/908e95f28cab4170b5a1c00be4283d1a

**Review by sojan-official — COMMENTED** [review]
code looks good to me
minor suggestions on the test cases.

File: spec/models/conversation_spec.rb [hunk 1]
```diff
@@ -621,4 +621,44 @@
...
+
+    it 'Sort conversations based on created_at' do
+      records = described_class.sort_on_created_at
+
+      expect(records.first.id).to eq(conversation_4.id)
+      expect(records.last.id).to eq(conversation_2.id)
+    end
+
+    it 'Sort conversations based on last_user_message_at' do
+      records = described_class.last_user_message_at
```
[the remark this answers is not in the document]
Comment (tejaswinichile) [reply spec/models/conversation_spec.rb:651]: We have conversation_1 without the message, let me add `expectation` for the same

File: spec/models/conversation_spec.rb [hunk 2]
```diff
@@ -621,4 +621,57 @@
...
+
+      it 'Sort conversations with latest resolved conversation at first' do
+        conversation_4.resolved!
+        records = described_class.latest
+
+        expect(records.first.id).to eq(conversation_4.id)
+        expect(records.last.id).to eq(conversation_2.id)
+      end
+
+      it 'Sort conversations with latest mes' do
```
Comment (sojan-official) [inline spec/models/conversation_spec.rb:669]: `message` :D

**Review by sojan-official — DISMISSED** [review]
LGTM

**Review by aswindevps — APPROVED** [review]

**Pull request merged**